src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -192,6 +192,10 @@ public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDep
         {
             DefType defType = _type.GetClosestDefType();
 
+            // If we're producing a full vtable, none of the dependencies are conditional.
```

Comment:
Nit: If we add other conditional static dependencies in the future, wrapping the code below in an if check for `ShouldProduceFullVTable` will be friendlier to future developers.